libpod/kube.go

```diff
@@ -0,0 +1,277 @@
+package libpod
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containers/libpod/pkg/lookup"
+	"github.com/containers/libpod/pkg/rootless"
+	"github.com/containers/libpod/pkg/util"
+	"github.com/cri-o/ocicni/pkg/ocicni"
+	"github.com/ghodss/yaml"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	// ErrKubeNotSupported represents an error where we cannot populate a kube required field
+	ErrKubeNotSupported = errors.New("This libpod function is not supported in kube translation yet")
+)
+
+// InspectforKube takes a slice of libpod containers and generates
+// one v1.Pod description that includes just a single container.
+func InspectForKube(containers []*Container) error {
+	var output []byte
+	developmentComment := []byte("# Generation of Kubenetes YAML is still under development!\n")
+	logrus.Warn("This function is still under heavy development.")
+	// Iterate each container
+	for _, ctr := range containers {
+		// Generate the v1.Pod yaml description
+		pod, err := simplePodWithV1Container(ctr)
+		if err != nil {
+			return err
+		}
+		// Marshall the results
+		b, err := yaml.Marshal(pod)
+		if err != nil {
+			return err
+		}
+		output = append(output, developmentComment...)
+		output = append(output, b...)
+		// Output the v1.Pod with the v1.Container
+		fmt.Println(string(output))
+	}
+	return nil
+}
+
+// simplePodWithV1Container is a function used by inspect when kube yaml needs to be generated
+// for a single container.  we "insert" that container description in a pod.
+func simplePodWithV1Container(ctr *Container) (*v1.Pod, error) {
+	var containers []v1.Container
+	result, err := ContainerToV1Container(ctr)
+	if err != nil {
+		return nil, err
+	}
+	containers = append(containers, result)
+
+	tm := v12.TypeMeta{
+		Kind:       "Pod",
+		APIVersion: "v1",
+	}
+
+	// Add a label called "app" with the containers name as a value
+	labels := make(map[string]string)
+	labels["app"] = removeUnderscores(ctr.Name())
+	om := v12.ObjectMeta{
```

Comment:
Before crawling back into my hole, I'll just note that I'm not at all a fan of variables with less that 6 chars.  Especially so when the variable is part of other words.  Makes it hard to scan/search for them.  Granted a small function here, just noting.  I now crawl back into my hole.

Reply:
/me offers to shovel dirt into your hole